Mathlib/Algebra/BigOperators/Pi.lean

```diff
@@ -57,6 +58,26 @@ theorem pi_eq_sum_univ {ι : Type*} [Fintype ι] [DecidableEq ι] {R : Type*} [S
   ext
   simp
 
+section CommSemiring
+variable [CommSemiring α]
+
+lemma prod_indicator_apply (s : Finset ι) (f : ι → Set κ) (g : ι → κ → α) (j : κ) :
+    ∏ i ∈ s, (f i).indicator (g i) j = (s.inf f).indicator (∏ i ∈ s, g i) j := by
```

Comment:
Why not replace `s.inf f` by `⋂ x ∈ s, f x`?

Reply:
because that's how I use it in #17957, and also because `s` is a finset, so might as well

Reply:
Hmm, I'm not entirely convinced by this reasoning - the iInter spelling is probably more readable and easier to work with. What do you think of moving the application of `Finset.inf_set_eq_iInter` there rather than here?

Reply:
I don't have a strong opinion, just a hunch that my spelling is better. Will change over for now